### ISSUE TITLE
Behaviour of list-dependencies and dot

### ISSUE THREAD

**Issue by tmcdonell** [commercialhaskell/stack#3530]
### General summary

I was trying to use `stack list-dependencies` and ran into the problem that it was looking for a version of the hoopl library which had not been uploaded to Hackage. The same problem occurred with `stack dot`. I opened a ticket for that at haskell/hoopl#43 and it has since been uploaded.

This was using ghc-8.2.1 obtained via `stack setup` (nightly-2017-10-23). hoopl is a core library that comes with ghc.

As @hvr [suggested](https://github.com/haskell/hoopl/issues/43#issuecomment-340247044) in that ticket:

> @tmcdonell You should file a bug to those upstream tools, as they're working under wrong assumptions we don't guarantee.

Hopefully they can clarify if you have any further questions.

### Steps to reproduce

Possibly difficult to reproduce now that haskell/hoopl#43 has been resolved.

### Expected

List of library dependencies.

### Actual

```
$ stack list-dependencies
Didn't see hoopl-3.10.2.2 in your package indices.
Updating and trying again.
Selected mirror https://s3.amazonaws.com/hackage.fpcomplete.com/
Downloading timestamp
No updates to your package list were found
Updating package index Hackage (mirrored at https://s3.amazonaws.com/hackage.fpcomplete.com/) ...The following package identifiers were not found in your indices: hoopl-3.10.2.2
Possible candidates: hoopl-3.10.0.0, hoopl-3.10.0.1, hoopl-3.10.0.2, hoopl-3.10.1.0, hoopl-3.10.2.0, hoopl-3.10.2.1.
```

### Stack version

```
Version 1.5.1, Git revision 600c1f01435a10d127938709556c1682ecfd694e (4861 commits) x86_64 hpack-0.17.1
```

### Method of installation

* Official binary, downloaded from http://www.haskellstack.org

**Comment by tmcdonell** [comment]
FWIW, I found a different example:

### Steps to reproduce

```bash
$ git clone https://github.com/tmcdonell/accelerate-blas.git
$ cd accelerate-blas
$ stack --stack-yaml=stack-7.10.yaml list-dependencies
Didn't see bin-package-db-0.0.0.0 in your package indices.
Updating and trying again.
Selected mirror https://s3.amazonaws.com/hackage.fpcomplete.com/
Downloading timestamp
No updates to your package list were found
Updating package index Hackage (mirrored at https://s3.amazonaws.com/hackage.fpcomplete.com/) ...The following package identifiers were not found in your indices: bin-package-db-0.0.0.0
```

Again, this is one of the core libraries:

```bash
$ stack --stack-yaml=stack-7.10.yaml exec ghc-pkg -- list
/Users/trevor/.stack/programs/x86_64-osx/ghc-7.10.3/lib/ghc-7.10.3/package.conf.d
   Cabal-1.22.5.0
   array-0.5.1.0
   base-4.8.2.0
   bin-package-db-0.0.0.0
   binary-0.7.5.0
   bytestring-0.10.6.0
   containers-0.5.6.2
   deepseq-1.4.1.1
   directory-1.2.2.0
   filepath-1.4.0.0
   ghc-7.10.3
   ghc-prim-0.4.0.0
   haskeline-0.7.2.1
   hoopl-3.10.0.2
   hpc-0.6.0.2
   integer-gmp-1.0.0.0
   pretty-1.1.2.0
   process-1.2.3.0
   rts-1.0
   template-haskell-2.10.0.0
   terminfo-0.4.0.1
   time-1.5.0.1
   transformers-0.4.2.0
   unix-2.7.1.0
   xhtml-3000.2.1
```

**Comment by mgsloan** [comment]
@tmcdonell Can you please try with the latest master / the 1.6 prerelease binaries? I believe this is fixed.  If it doesn't work with that you please create a github repo that reproduces the issue?  I've spent the time to try both methods of reproduction, but neither depending on hoop-3.10.2.2 nor running this on accelerate-blas is causing issues:

```
mgsloan@computer:~/fpco/test-stack/issue3530/hmm$ stack list-dependencies
base 4.9.1.0
containers 0.5.7.1
ghc-prim 0.5.0.0
hmm 0.1.0.0
hoopl 3.10.2.2
integer-gmp 1.0.0.1
rts 1.0
```

```
mgsloan@computer:~/fpco/test-stack/issue3530/accelerate-blas$ stack --stack-yaml=stack-7.10.yaml list-dependencies
Cabal 2.0.0.2
abstract-deque 0.3
accelerate 1.1.1.0
accelerate-blas 0.1.0.1
accelerate-llvm 1.1.0.0
accelerate-llvm-native 1.1.0.1
accelerate-llvm-ptx 1.1.0.0
ansi-terminal 0.6.2.3
ansi-wl-pprint 0.6.7.3
array 0.5.1.0
...
```

**Comment by tmcdonell** [comment]
Ah, this indeed appears to be fixed in the 1.6 prerelease (or at least, works with all the examples I have). Apologies for the noise!

```
> stack --version
Version 1.6.0.20171022, Git revision 7bddfaf7f9f8cd9ec1c710fa83e77262e494eee4 (5285 commits) x86_64 hpack-0.18.1
```
